File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/dettagli/DettaglioFertilizzazione.ts

import { Effluente } from 'app/Model/metaschema/Effluente';
import { TipoFertilizzante } from 'app/Model/metaschema/TipoFertilizzante';
import { TipologiaFertilizzante } from 'app/Model/metaschema/TipologiaFertilizzante';
import { RisorsaProdotto } from '../risorse/RisorsaProdotto';

export class DettaglioFertilizzazione extends RisorsaProdotto {
    N: number;
    P: number;
    K: number;
    Cu: number;
    Mg: number;
    efficienza: number;
    effluente: Effluente;
    tipoFertilizzante: TipoFertilizzante;
    tipologieFertilizzante: TipologiaFertilizzante[];
    N_Ponderato: boolean;
    P_Ponderato: boolean;
    K_Ponderato: boolean;
    Cu_Ponderato: boolean;

    constructor() {
        super();
        this.classType = 'DettaglioFertilizzazione';
    }
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/dettagli/DettaglioFertilizzazione.cs
using AgronicaCoreModelsSTD.attivita.risorse;
using System.Collections.Generic;
using System;
using AgronicaCoreModelsSTD.metaschema;
using Newtonsoft.Json;

namespace AgronicaCoreModelsSTD.attivita.dettagli
{

    public class DettaglioFertilizzazione : RisorsaProdotto
    {
        public decimal N { get; set; }
        public decimal P { get; set; }
        public decimal K { get; set; }
        public decimal Cu { get; set; }
        public decimal Mg { get; set; }

        public decimal efficienza { get; set; }

        public Effluente effluente { get; set; }

        public TipoFertilizzante tipoFertilizzante { get; set; }

        public List<TipologiaFertilizzante> tipologieFertilizzante { get; set; }

        public bool N_Ponderato { get; set; }
        public bool P_Ponderato { get; set; }
        public bool K_Ponderato { get; set; }
        public bool Cu_Ponderato { get; set; }


        public DettaglioFertilizzazione()
        {
            classType = costanti.ClassType.DettaglioFertilizzazione;
        }

        public new DettaglioFertilizzazione Clona()
        {
            try
            {

                string output = JsonConvert.SerializeObject(this);
                DettaglioFertilizzazione deserializedObject = JsonConvert.DeserializeObject<DettaglioFertilizzazione>(output);
                return deserializedObject;
            }
            catch (Exception ex)
            {
                throw new Exception("DettaglioFertilizzazione.Clona: " + ex.Message);
            }

        }

    }
}
